src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -752,7 +752,7 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
             && !isConvention(s)
           )
         for (s <- unusedPrivates.unusedParams if warnable(s))
-          emitUnusedWarning(s.pos, s"parameter $s in ${if (s.owner.isAnonymousFunction) "anonymous function" else s.owner} is never used", WarningCategory.UnusedParams, s)
+          emitUnusedWarning(s.pos, s"parameter $s in ${if (s.owner.isAnonymousFunction) "anonymous function" else s.owner} is never used\nType: ${s.tpe.toLongString}", WarningCategory.UnusedParams, s)
```

Comment:
is `toLongString` the best way to show complete type information here?  or `nameAndArgsString`?